src/throttle.rs

```diff
@@ -234,14 +259,93 @@ fn usize_to_u32_saturate(x: usize) -> u32 {
     x.try_into().unwrap_or(u32::MAX)
 }
 
+enum ThrottledUploadState {
+    Idle,
+    Sleeping(Pin<Box<dyn Future<Output = ()> + Send>>),
+    Waiting,
+}
+
+struct ThrottledUpload {
```

Comment:
As far as I can tell this simply delays the first write or flush, I'm not really sure this is particularly meaningful. As this cannot delay the underlying API calls (as they're hidden behind the AsyncWrite abstraction), I would personally be inclined to just not support these APIs in the ThrottledStore

Reply:
Is `ThrottledStore` mostly used just for testing?

Reply:
Solely used for testing

Reply:
I've removed the ThrottleStore implementation.